### PR TITLE
fix: use a button element for clickable icons

### LINKED ISSUE #10265: [Bug Report] VTextField append icons are not focusable
### Environment
**Vuetify Version:** 2.2.4
**Vue Version:** 2.6.11
**Browsers:** Chrome 79.0.3945.117
**OS:** Windows 10

### Steps to reproduce
1. Focus on the text field
2. Try to press tab or shift tab

### Expected Behavior
The append/prepend icons should get focus and have an active focus state

### Actual Behavior
They don't

### Reproduction Link
<a href="https://codepen.io/phiter/pen/KKwGyWR" target="_blank">https://codepen.io/phiter/pen/KKwGyWR</a>

### LINKED ISSUE #10391: [Bug Report] "@click:append" on VAutocomplete removes aria-hidden=true
### Environment
**Vuetify Version:** 2.2.8
**Vue Version:** 2.6.11
**Browsers:** Chrome 79.0.3945.130
**OS:** Windows 10

### Steps to reproduce
Click into "DropDown Focus Bug" Input field and click "Tab" to move focus to next control

### Expected Behavior
Focus jumps to next input field

### Actual Behavior
DropDown Icon of Input Field gets the focus (because of @click:append="console.log('click')")

### Reproduction Link
<a href="https://codepen.io/mseele/pen/NWPZWEV" target="_blank">https://codepen.io/mseele/pen/NWPZWEV</a>

### Other comments
Please see the second Input Field with out @click:append="console.log('click')" where the issue is gone

### LINKED ISSUE #10398: [Bug Report] eye icon on textfield does not work on enter
### Environment
**Vuetify Version:** 2.2.8
**Vue Version:** 2.6.11
**Browsers:** Chrome 79.0.3945.130
**OS:** Windows 10

### Steps to reproduce
tab through password field using tabulator
stop at icon
press enter

### Expected Behavior
it should show/hide password

### Actual Behavior
does nothing, only working on click

### Reproduction Link
<a href="https://codepen.io/rveruna/pen/rNaEJzZ" target="_blank">https://codepen.io/rveruna/pen/rNaEJzZ</a>

### Other comments
you can see that on your documentation page as well

### PR DESCRIPTION
**Pull request by KaelWD**
## Description

fixes #10265
fixes #10391
fixes #10398
Icon color specs: https://material.io/design/iconography/system-icons.html#color

## Markup:

<details>

```vue
<template>
  <div class="ma-12 pa-12">
    <v-container fluid>
      <v-row>

        <v-col cols="12">
          <v-rating></v-rating>
        </v-col>

        <v-col cols="12">
          <v-autocomplete
            v-model="value"
            :items="items"
            @click:append="console.log('click')"
            dense
            filled
            label="DropDown Focus Bug"
          ></v-autocomplete>
        </v-col>

        <v-col
          cols="12"
          sm="6"
        >
          <v-text-field
            disabled
            :append-icon="mdiVuetify"
            label="Foobar"
            @click:append="show1 = !show1"
          ></v-text-field>
        </v-col>

        <v-col
          cols="12"
          sm="6"
        >
          <v-icon
            @click="onClick"
            v-text="mdiVuetify"
          ></v-icon>
          <v-text-field
            v-model="password"
            :append-outer-icon="show1 ? 'mdi-eye' : 'mdi-eye-off'"
            :rules="[rules.required, rules.min]"
            :type="show1 ? 'text' : 'password'"
            name="input-10-1"
            label="Normal with hint text"
            hint="At least 8 characters"
            counter
            @click:append-outer="onClick"
          ></v-text-field>
        </v-col>

        <v-col
          cols="12"
          sm="6"
        >
          <v-text-field
            :append-icon="show2 ? 'mdi-eye' : 'mdi-eye-off'"
            :rules="[rules.required, rules.min]"
            :type="show2 ? 'text' : 'password'"
            name="input-10-2"
            label="Visible"
            hint="At least 8 characters"
            value="wqfasds"
            class="input-group--focused"
            @click:append="show2 = !show2"
          ></v-text-field>
        </v-col>
        <v-col
          cols="12"
          sm="6"
        >
          <v-text-field
            :append-icon="show3 ? 'mdi-eye' : 'mdi-eye-off'"
            :rules="[rules.required, rules.min]"
            :type="show3 ? 'text' : 'password'"
            name="input-10-2"
            label="Not visible"
            hint="At least 8 characters"
            value="wqfasds"
            class="input-group--focused"
            @click:append="show3 = !show3"
          ></v-text-field>
        </v-col>
        <v-col
          cols="12"
          sm="6"
        >
          <v-text-field
            :append-icon="show4 ? 'mdi-eye' : 'mdi-eye-off'"
            :rules="[rules.required, rules.emailMatch]"
            :type="show4 ? 'text' : 'password'"
            name="input-10-2"
            label="Error"
            hint="At least 8 characters"
            value="Pa"
            error
            @click:append="show4 = !show4"
          ></v-text-field>
        </v-col>

      </v-row>
    </v-container>
  </div>
</template>

<script>
  // Icons
  import {
    mdiVuetify,
  } from '@mdi/js'
  export default {
    data () {
      return {
        mdiVuetify,
        value: null,
        items: ['foo', 'bar', 'fizz', 'buzz'],
        show1: false,
        show2: true,
        show3: false,
        show4: false,
        password: 'Password',
        rules: {
          required: value => !!value || 'Required.',
          min: v => v.length >= 8 || 'Min 8 characters',
          emailMatch: () => ('The email and password you entered don\'t match'),
        },
        console,
      }
    },
    methods: {
      onClick (e) {
        // e.preventDefault()
        // e.stopPropagation()
        // e.returnValue = false
        // e.cancel = true
        this.show1 = !this.show1
        console.log('here 1')
      },
    },
  }
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

### CODE REVIEW

File: packages/vuetify/src/components/VIcon/VIcon.sass [hunk 1]
```diff
@@ -6,9 +6,6 @@
   color: map-deep-get($material, 'text-color')
   opacity: map-deep-get($material, 'icons', 'active')
 
-  &:focus
-    opacity: map-deep-get($material, 'icons', 'focused')
```
Comment (KaelWD) [inline packages/vuetify/src/components/VIcon/VIcon.sass]: image

Comment (KaelWD) [reply packages/vuetify/src/components/VIcon/VIcon.sass]: Also v-rating now has a double ripple
image

Comment (johnleider) [reply packages/vuetify/src/components/VIcon/VIcon.sass]: Might be able to remove the need for v-btn in that component, looking into it.

Comment (johnleider) [reply packages/vuetify/src/components/VIcon/VIcon.sass]: Also there was an overwhelming majority for not having both.

**Pull request merged**